modules/mergeImageWithTelemetry/mergeImageWithTelemetryWorker.py

```diff
@@ -0,0 +1,45 @@
+from modules.mergeImageWithTelemetry.mergeImageWithTelemetry import MergeImageWithTelemetry
+
+def pipelineMergeWorker(pause, exitRequest, imagePipelineIn, telemetryPipelineIn, pipelineOut):
+    print("start pipeline merge")
+    mergeImageWithTelemetry = MergeImageWithTelemetry()
+    shouldGetImage = True
+    curImage = None
+
+    while True: 
+        if not exitRequest.empty():
+            return
+        
+        pause.acquire()
+        pause.release()
+        
+        if shouldGetImage:
+            curImage = imagePipelineIn.get()
+
+        if curImage is None:
+            continue
+
+        # put current elements of the telemetry pipeline into the queue
+        while not telemetryPipelineIn.empty(): 
```

Comment:
This has the potential to be an infinite loop, the telemetryPipeline will keep getting filled faster than this can pull, instead, each run should just get the latest data in the telemetry pipeline. Essentially, each time the **parent** loop runs, it should get the latest telemetry, add it to the telemtryData array, get the current image,and matches the current image with the correct telemetry Data. 

Because the decklinksrc program will almost certainly run slower than the command module, we know that whatever image we have at that point is certainly older than the item at the head of the telemetry pipeline (assuming LIFO). Because we add the newest data to the tail of the pipeline, we logically know that any data older than the item at the head of the telemetry pipeline is already in the telemetryData array. We know that the telemetry data that matches the image will be older than the item at the head of the telemtry pipeline, therefore we can deduce that by implementing the change in the comment above, the data with a timestamp closest to the current image is in telemetryData somewhere. 

I wish I could write this in clean mathematical language but not possible here, we can discuss this further as well.